bazel/linux/qemu.bzl

```diff
@@ -38,33 +45,45 @@ test -n "$QEMU_BINARY" || {{
     exit 10
 }}
 
-if [ -n "$ROOTFS" ]; then
-    QEMU_FLAGS+=("-drive" "file=$ROOTFS,if=virtio,cache=none")
-else
-    QEMU_FLAGS+=("-fsdev" "local,security_model=none,multidevs=remap,id=fsdev-fsRoot,path=/")
-    QEMU_FLAGS+=("-device" "virtio-9p-pci,fsdev=fsdev-fsRoot,mount_tag=/dev/root")
-    KERNEL_FLAGS+=("root=/dev/root" "rootfstype=9p" "init=$INIT")
-    KERNEL_FLAGS+=("rootflags=trans=virtio,version=9p2000.L,msize=5000000,cache=mmap,posixacl")
+if [[ "$ARCH" == "x86" ]]; then
+    if [ -n "$ROOTFS" ]; then
+        QEMU_FLAGS+=("-drive" "file=$ROOTFS,if=virtio,cache=none")
+    else
+        QEMU_FLAGS+=("-fsdev" "local,security_model=none,multidevs=remap,id=fsdev-fsRoot,path=/")
+        QEMU_FLAGS+=("-device" "virtio-9p-pci,fsdev=fsdev-fsRoot,mount_tag=/dev/root")
+        KERNEL_FLAGS+=("root=/dev/root" "rootfstype=9p" "init=$INIT")
+        KERNEL_FLAGS+=("rootflags=trans=virtio,version=9p2000.L,msize=5000000,cache=mmap,posixacl")
+    fi
+
+    QEMU_FLAGS+=("-fsdev" "local,security_model=none,multidevs=remap,id=fsdev-fsOutputDir,path=$OUTPUT_DIR")
+    QEMU_FLAGS+=("-device" "virtio-9p-pci,fsdev=fsdev-fsOutputDir,mount_tag=/dev/output_dir")
+
+    test -z "$KERNEL" || QEMU_FLAGS+=("-kernel" "$KERNEL")
+    test -z "$SINGLE" || KERNEL_FLAGS+=("init=/bin/sh")
 fi
 
-QEMU_FLAGS+=("-fsdev" "local,security_model=none,multidevs=remap,id=fsdev-fsOutputDir,path=$OUTPUT_DIR")
-QEMU_FLAGS+=("-device" "virtio-9p-pci,fsdev=fsdev-fsOutputDir,mount_tag=/dev/output_dir")
+if [[ "$ARCH" == "aarch64" ]] ; then
+    RAW_BOOTRAM_LOADER="-device loader,file=$BOOTRAM_IMAGE,addr=0"
+    ITB_IMAGE_LOADER="-device loader,file=$KERNEL,addr=0x10000000000"
 
-test -z "$KERNEL" || QEMU_FLAGS+=("-kernel" "$KERNEL")
-test -z "$SINGLE" || KERNEL_FLAGS+=("init=/bin/sh")
+    QEMU_FLAGS+="$RAW_BOOTRAM_LOADER $ITB_IMAGE_LOADER"
+fi
+
+WRAPPER_OPTS+="arch=$ARCH"
 
-QEMU_FLAGS+=("-append" "${{KERNEL_FLAGS[*]}} ${{KERNEL_OPTS[*]}}")
 QEMU_FLAGS+=("${{EMULATOR_OPTS[@]}}")
 
 if [ ${{#WRAPPER_OPTS[@]}} -ne 0 ]; then
     WRAPPER_OPTS=('--' "${{WRAPPER_OPTS[@]}}")
 fi
 
+# Pass qemu/kernel/wrapper opts as three separate args to Qemu binary.
+
 echo 1>&2 '$' "$QEMU_BINARY" "${{QEMU_FLAGS[@]}}" "${{WRAPPER_OPTS[@]}}"
 if [ -z "$INTERACTIVE" -a -z "$SINGLE" ]; then
-    "$QEMU_BINARY" "${{QEMU_FLAGS[@]}}" "${{WRAPPER_OPTS[@]}}" </dev/null | tee "$OUTPUT_FILE"
+    "$QEMU_BINARY" "${{QEMU_FLAGS[*]}}"  "${{KERNEL_FLAGS[*]}}" "${{WRAPPER_OPTS[*]}}" </dev/null | tee "$OUTPUT_FILE"
```

Comment:
The original code had:
```
QEMU_FLAGS+=("-append" "${{KERNEL_FLAGS[*]}} ${{KERNEL_OPTS[*]}}")
```

... what happened to KERNEL_OPTS in the new code? Can't find anywhere the -append, and handling of KERNEL_OPTS?

Reply:
@ccontavalli  Very sorry.  Seems like I dropped KERNEL_OPTS when I mucked with args.  As for -append I pass all the args (kernel/qemu and wrapper args) separately to the model-wrapper-socket script in internal repo and do the append there (a different PR https://github.com/enfabrica/internal/pull/40251).  

The change is at 
https://github.com/enfabrica/internal/pull/40251/commits/3e8fa4754bb040fae9bb968e9e04371ef6eb6151#diff-2d20aee07523ca136cf9db6baeb3534f46d4c33e6bb9add34e88a9f63821f0eaR87


Any way thinking again,  I went back the old way of doing things since if I do the above I realized there are other model-wrapper scripts which need to be changed too.  So I just pass the arguments similar to how we use to do so before.  

Also I just use the machine name now to know if the qemu to use is aarch64 or x86 one, instead of passing it as a wrapper option.  That way do not have to muck with wrapper opts.   If you rather have the arch passed as a wapper opts, I can change that.  Please let me know. 

Also to make it easy to read I am force pushing the diff.  Please take a look again.  Sorry again for the mess-up and thanks for finding the issue.